Prox/Prox/Utilities/OpenInHelper.swift

```diff
@@ -0,0 +1,168 @@
+            return true
+        }
+        print("Unable to open directions")
+        return false
+    }
+
+    fileprivate static func gmapsAppURLForRoute(fromLocation: CLLocationCoordinate2D, toLocation: CLLocationCoordinate2D, by transportType: MKDirectionsTransportType) -> URL? {
+        let directionsMode: String
+        switch transportType {
+        case MKDirectionsTransportType.automobile:
+            directionsMode = "driving"
+        case MKDirectionsTransportType.transit:
+            directionsMode = "transit"
+        case MKDirectionsTransportType.walking:
+            directionsMode = "walking"
+        default:
+            directionsMode = ""
+            return nil
+        }
+        let queryParams = ["saddr=\(fromLocation.latitude),\(fromLocation.longitude)", "daddr=\(toLocation.latitude),\(toLocation.longitude)", "directionsMode=\(directionsMode)"]
+
+        let gmapsRoutingRequestURLString = gmapsAppSchemeString + "?" + queryParams.joined(separator: "&")
+        return URL(string: gmapsRoutingRequestURLString)
+    }
+
+
+    fileprivate static func gmapsWebURLForRoute(fromLocation: CLLocationCoordinate2D, toLocation: CLLocationCoordinate2D, by transportType: MKDirectionsTransportType) -> URL? {
+        guard let dirFlg = dirFlgForTransportType(transportType: transportType) else {
+            return nil
+        }
+        let queryParams = ["saddr=\(fromLocation.latitude),\(fromLocation.longitude)", "daddr=\(toLocation.latitude),\(toLocation.longitude)", "dirflg=\(dirFlg)"]
+
+        let gmapsRoutingRequestURLString = gmapsWebSchemeString + "?" + queryParams.joined(separator: "&")
+        return URL(string: gmapsRoutingRequestURLString)
+    }
+
+
+    fileprivate static func appleMapsURLForRoute(fromLocation: CLLocationCoordinate2D, toAddress: String, by transportType: MKDirectionsTransportType) -> URL? {
+        guard let dirFlg = dirFlgForTransportType(transportType: transportType) else {
+            return nil
+        }
+
+        let queryParams = ["daddr=\(encodeByAddingPercentEscapes(toAddress))", "dirflg=\(dirFlg)"]
+
+        let appleMapsRoutingRequestURLString = appleMapsSchemeString + "?" + queryParams.joined(separator: "&")
+        return URL(string: appleMapsRoutingRequestURLString)
+    }
+
+    fileprivate static func dirFlgForTransportType(transportType: MKDirectionsTransportType) -> String? {
+        switch transportType {
+        case MKDirectionsTransportType.automobile:
+            return "d"
+        case MKDirectionsTransportType.transit:
+            return "r"
+        case MKDirectionsTransportType.walking:
+            return "w"
+        default:
+            return nil
+        }
+    }
+
+    //MARK: Helper functions
+    fileprivate static func encodeByAddingPercentEscapes(_ input: String) -> String {
+        return NSString(string: input).addingPercentEncoding(withAllowedCharacters: CharacterSet(charactersIn: "!*'();:@&=+$,/?%#[]"))!
```

Comment:
We also do this in the Firefox helper – can we put the code somewhere central to avoid duplication?